contrib/world_builder/include/world_builder/coordinate_systems/interface.h

```diff
@@ -0,0 +1,93 @@
+/*
+  Copyright (C) 2018 by the authors of the World Builder code.
+
+  This file is part of the World Builder.
+
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU Lesser General Public License as published
+   by the Free Software Foundation, either version 2 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public License
+   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef _world_builder_coordinate_systems_interface_h
+#define _world_builder_coordinate_systems_interface_h
+
+#include <boost/property_tree/ptree.hpp>
+
+#include <world_builder/utilities.h>
+#include <world_builder/coordinate_system.h>
+
+
+using boost::property_tree::ptree;
+
+namespace WorldBuilder
+{
+  namespace CoordinateSystems
+  {
+
+    class Interface
+    {
+      public:
+        /**
+         * constructor
+         */
+        Interface();
+
+        /**
+         * Destructor
+         */
+        ~Interface();
```

Comment:
all interface classes in c++ should have a virtual destructor. https://stackoverflow.com/a/461224